src/app/chart/chart.component.ts

```diff
@@ -1,47 +1,42 @@
-import {
-  Component,
-  Input,
-  OnChanges,
-  SimpleChanges,
-  OnInit,
-} from '@angular/core';
+import { Component, Input } from '@angular/core';
 import { MockDataService } from '../mockData/Table data';
+import {
+  TableDataInterface,
+  ChartData,
+  OrderSummary,
+} from '../shared/interfaces';
 
 @Component({
   selector: 'app-chart',
   templateUrl: './chart.component.html',
   styleUrls: ['./chart.component.scss'],
 })
-export class ChartComponent implements OnInit, OnChanges {
-  @Input() data: any[] = [];
-  ordersData: any;
-  chartData: any;
-
-  constructor(public mockDataService: MockDataService) {}
+export class ChartComponent {
+  private _data: TableDataInterface[] = [];
+  ordersData: OrderSummary[] = [];
+  chartData: ChartData[] = [];
 
-  ngOnInit(): void {
+  @Input()
+  set data(value: TableDataInterface[]) {
+    this._data = value;
     this.updateChartData();
   }
 
-  ngOnChanges(changes: SimpleChanges): void {
-    if (changes['data'] && !changes['data'].firstChange) {
-      this.updateChartData();
-    }
-  }
+  constructor(public mockDataService: MockDataService) {}
 
   updateChartData(): void {
-    this.ordersData = this.data
-      .filter((order: any) => order.orderId && order.orderPieces)
-      .map((order: any) => ({
+    this.ordersData = this._data
+      .filter((order: TableDataInterface) => order.orderId && order.orderPieces)
+      .map((order: TableDataInterface) => ({
         orderId: order.orderId,
         orderPieces: order.orderPieces,
         customerName: order.customerName,
-      }));
+      })) as OrderSummary[];
 
     this.chartData = this.getChartData();
   }
 
-  getChartData() {
+  getChartData(): ChartData[] {
```

Comment:
Please refactor this method introducing a ranges array to define the range boundaries. Then iterate over this array to generate each chart data entry dynamically. This approach will reduce code repetition, improve maintainability and make future extensions simpler.